Federation: create loadbalancer service in tests only if test depends on it

shashidharatd opened the pull request:
**What this PR does / why we need it**:
Creating LoadBalancer type of service for every test case is kind of expensive and time consuming to provision. So this PR changes the test cases to use LoadBalancer type services only when necessary.

**Which issue this PR fixes** (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged): fixes #47068 

**Release note**:
```release-note
NONE
```
cc @kubernetes/sig-federation-pr-reviews 
/assign @madhusudancs

Commit:
create loadbalancer service in tests only if test depends on it

marun commented on test/e2e_federation/util.go:
```diff
@@ -209,9 +212,11 @@ func cleanupServiceShardsAndProviderResources(namespace string, service *v1.Serv
+		if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+			err = cleanupServiceShardLoadBalancer(name, cSvc, fedframework.FederatedDefaultTestTimeout)
+			if err != nil {
+				framework.Logf("Failed to delete cloud provider resources for service %q in namespace %q, in cluster %q", service.Name, namespace, name)
```
Is there a reason not to log the error?

shashidharatd replied:
agree, always better to log err. updated now.

marun commented on test/e2e_federation/authn.go:
```diff
@@ -44,7 +45,7 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 
 			nsName := f.FederationNamespace.Name
-			svc := createServiceOrFail(fcs, nsName, FederatedServiceName)
+			svc := createServiceOrFail(fcs, nsName, FederatedServiceName, v1.ServiceTypeClusterIP)
```
(No action required) It there are only 2 types of services that are in use, it might be cleaner to have 2 separate methods (that both call an underlying method with the appropriate argument)>

```golang
createServiceOrFail(...)
createLBServiceOrFail(...)

```

shashidharatd replied:
Agree with your suggestion. its much cleaner. updated accordingly.

marun commented:
A nit about logging but otherwise I think this is uncontroversial.

shashidharatd commented:
pull-kubernetes-e2e-kops-aws and pull-kubernetes-verify are also failing consistently now.
Need to wait for their resolution.

shashidharatd commented:
@marun, have addressed your comments now. PTAL.
@madhusudancs, this PR reduces the reliance of federation e2e tests on LB services, thereby giving little relief from the issue which is described in https://github.com/kubernetes/kubernetes/issues/46859

Can we prioritize this and apply milestone. Thanks !

shashidharatd commented:
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

shashidharatd commented:
@k8s-bot pull-kubernetes-e2e-kops-aws test this

shashidharatd commented:
@k8s-bot pull-kubernetes-e2e-kops-aws test this

shashidharatd commented:
@k8s-bot pull-kubernetes-verify test this

cblecker commented:
/priority failing-test

madhusudancs commented:
@shashidharatd thanks!

/lgtm

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/46120#-:-Klz5jPd9xeTXBn2xPem:b-q1sjke)*
<!-- Sent from Reviewable.io -->

madhusudancs commented:
/approve

madhusudancs commented:
@kubernetes/kubernetes-release-managers this is a test improvement.

shashidharatd commented:
/retest

shashidharatd commented:
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

ericchiang commented:
/retest

shashidharatd commented:
/retest
